core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.ts

```diff
@@ -240,23 +256,46 @@ export class ContributorAdminDashboardPageComponent implements OnInit {
     this.languageDropdownShown = !this.languageDropdownShown;
   }
 
-  toggleActivityDropdown(): void {
-    this.activityDropdownShown = !this.activityDropdownShown;
+  getDateThatIsDaysBeforeToday(numberOfDaysBeforeToday: number): Date {
+    const today = new Date();
+    return new Date(
+      today.getTime() - this.ONE_DAY_IN_MILLIS * numberOfDaysBeforeToday
+    );
+  }
+
+  getNumberOfDaysForDateBeforeToday(date: Date): number {
+    const today = new Date();
+    return Math.floor(
+      (today.getTime() - new Date(date).getTime()) / this.ONE_DAY_IN_MILLIS
+    );
+  }
+
+  isValidLastDate(selectedLastDate: Date): boolean {
+    const today = new Date();
+    return today.getTime() >= selectedLastDate.getTime();
   }
 
   createFilter(): void {
     const tempFilter = new ContributorAdminDashboardFilter(
       this.selectedTopicIds,
       this.selectedLanguage.id,
       null,
-      this.selectedLastActivity
+      this.getNumberOfDaysForDateBeforeToday(this.lastDateToFilterUsersActivity)
     );
 
     if (this.filter === undefined || !isEqual(tempFilter, this.filter)) {
       this.filter = tempFilter;
     }
   }
 
+  changeLastDate(value: Date): void {
```

Comment:
Why this is required? Please add a comment.

Reply:
This function acts as a event handler for date picker. When ever we select a new date in the date picker it update the variable that stores date and create filter corresponding to the newly selected date.